ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/features/NeedRetryTest.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2014-2019 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.client.tests.features
+
+import io.ktor.client.engine.mock.*
+import io.ktor.client.features.*
+import io.ktor.client.request.*
+import io.ktor.client.response.*
+import io.ktor.client.tests.utils.*
+import kotlin.test.*
+
+class NeedRetryTest {
+
+    @Test
+    fun testNeedRetryCalled() = clientTest(MockEngine {
+        respondOk("Hello")
+    }) {
+
+        var retryConditionBlockCalled = false
+
+        config {
+            RetryCondition {
+                retryCondition { requestBuilder, response ->
+                    retryConditionBlockCalled = true
+                    false
+                }
+            }
+        }
+
+        test { client ->
+            client.get<HttpResponse>()
```

Comment:
Consider closing the response here.

Reply:
Btw. We need the test with different two responses for first and second requests to check that we're getting a proper response instance in the end.